chapters/intro/main.tex

```diff
@@ -32,22 +171,220 @@ \subsection{Code snippets}
+All of the test suites associated with this thesis achieve 100\% coverage.
+
+% TODO Should I mention property-based hypothesis testing?
+
+To regularly test code that is going to be merged into the main code base
+(through version control), continuous integration (CI) systems exist. CI systems
+run the test suite and coverage checks at regular prompts (e.g. when a new
+version is pushed to the online repository, prior to new releases of the
+software, according to a schedule, etc.), minimising any potential issues during
+development and collaboration as well as providing another layer of
+transparency. Given that the code for this thesis is hosted on GitHub, the CI
+used is GitHub Actions~\cite{github-actions}.
+
+\subsubsection{Documentation}
+
+In addition to testing, another crucial appendix to a software code base is its
+\emph{documentation}. Software documentation can take many forms --- text,
+websites, illustrations, demonstrations --- but regardless of how it is
+presented, the purpose is to explain to a user how to use a piece of software.
+
+All of the repositories associated with this thesis include (at a minimum) a
+\mintinline{console}{README} file, detailing what the repository is for, and (if
+appropriate) instructions on how to reproduce the results with the code therein.
+Each Python function, method and class defined in the source code includes its
+own inline documentation in the form of a \emph{docstring}. Furthermore, the
+variables and defined objects have been assigned informative, sensible names,
+making the software self-documenting.
+
+For the larger, free-standing software packages developed during this thesis,
+fully fledged documentation websites have been written. Each of these is hosted
+on \href{https://readthedocs.org/}{Read the Docs} and adheres to the so-called
+`Grand Unified Theory of Documentation'~\cite{documentation}, which separates
+documentation into four categories: tutorials, how-to guides, explanation and
+reference.
+
+\subsection{Summary of software}
+
+As stated throughout this section, the software to accompany this thesis has
+been written according to best practices, and their associated repositories are
+available online. These practices have been adopted to ensure the reliability,
+reproducibility and sustainability of the software described throughout this
+thesis.
+
+In addition to these GitHub repositories, the specific versions of the source
+code used in each chapter have been archived online via Zenodo~\cite{zenodo}.
+Each archive is assigned a digital object identifier (DOI) name, further
+reinforcing the longevity of the software. Table~\ref{tab:repos} details the
+repositories and archives associated with each chapter, where appropriate.
+
+\begin{table}[tbhp]
+    \centering%
+    \resizebox{\textwidth}{!}{%
+        \input{chapters/intro/tex/repos/main.tex}%
+    }\caption{%
+        A summary of the repositories and archives associated with the chapters
+        of this thesis%
```

Comment:
I wonder if you could add a sentence (or brackets) explaining when it's not appropriate (`due to patient data confidentiality`) or something like that?

Reply:
Yeah, I had thought that myself. Added a sentence in 8b77263.